agent_build/build_package.py

```diff
@@ -0,0 +1,1193 @@
+                    subprocess.check_call(
+                        ["docker", "save", image_name],
+                        stdout=f
+
+                    )
+
+    @classmethod
+    def dump_build_environment_files_content_checksum(
+            cls,
+            checksum_output_path: Union[str, pl.Path]
+    ):
+        """
+        Dump the checksum of the content of the file used during the 'prepare-build-environment' action.
+            For more info see 'dump-checksum' action in class docstring.
+
+        :param checksum_output_path: Is mainly created for the CI/CD purposes. If specified, the function dumps the
+            file with the checksum of all the content of all files which are used during the preparation
+            of the build environment. This checksum can be used by CI/CD as the cache key..
+        """
+        checksum = cls._get_build_environment_files_checksum()
+
+        checksum_output_path = pl.Path(checksum_output_path)
+        checksum_output_path.parent.mkdir(exist_ok=True, parents=True)
+        checksum_output_path.write_text(checksum)
+
+
+    @property
+    def _build_info(self) -> Optional[str]:
+        """Returns a string containing the package build info."""
+
+        build_info_buffer = io.StringIO()
+
+        # We need to execute the git command in the source root.
+        # Add in the e-mail address of the user building it.
+        try:
+            packager_email = subprocess.check_output(
+                "git config user.email", shell=True, cwd=str(__SOURCE_ROOT__)
+            ).decode().strip()
+        except subprocess.CalledProcessError:
+            packager_email = "unknown"
+
+        print("Packaged by: %s" % packager_email.strip(), file=build_info_buffer)
+
+        # Determine the last commit from the log.
+        commit_id = subprocess.check_output(
+                "git log --summary -1 | head -n 1 | cut -d ' ' -f 2",
+                shell=True,
+                cwd=__SOURCE_ROOT__,
+        ).decode().strip()
+
+        print("Latest commit: %s" % commit_id.strip(), file=build_info_buffer)
+
+        # Include the branch just for safety sake.
+        branch = subprocess.check_output(
+            "git branch | cut -d ' ' -f 2", shell=True, cwd=__SOURCE_ROOT__
+        ).decode().strip()
+        print("From branch: %s" % branch.strip(), file=build_info_buffer)
+
+        # Add a timestamp.
+        print(
+            "Build time: %s" % str(time.strftime("%Y-%m-%d %H:%M:%S UTC", time.gmtime())),
+            file=build_info_buffer,
+        )
+
+        return build_info_buffer.getvalue()
+
+    @staticmethod
+    def _add_config(
+            config_source_path: Union[str, pl.Path], output_path: Union[str, pl.Path]
+    ):
+        """
+        Copy config folder from the specified path to the target path.
+        """
+        config_source_path = pl.Path(config_source_path)
+        output_path = pl.Path(output_path)
```

Comment:
Do we also want to explicitly set permissions on this config directory + agent.json so it's not readable by others?